nimble/host/src/ble_hs.c

```diff
@@ -376,18 +375,12 @@ ble_hs_reset(void)
 
     ble_hs_clear_rx_queue();
 
-    while (1) {
-        conn_handle = ble_hs_atomic_first_conn_handle();
-        if (conn_handle == BLE_HS_CONN_HANDLE_NONE) {
-            break;
-        }
-
-        ble_gap_conn_broken(conn_handle, ble_hs_reset_reason);
-    }
-
     /* Clear configured addresses. */
     ble_hs_id_reset();
 
+    /* Clear adverising and scanning states. */
+    ble_gap_reset_state(ble_hs_reset_reason);
```

Comment:
I would call it before the `ble_hs_id_reset()`. It probably will change nothing, but since before we were calling `ble_gap_conn_broken()` before that I don't see a reason to change that.

Reply:
> It probably will change nothing

Had the same though in mind :smile: will make the changes.